troop/views.py

```diff
@@ -5,32 +5,36 @@
 from django.utils.translation import gettext_lazy as _
 from django.db.models import Count
 from django.conf import settings
+from django.core.exceptions import PermissionDenied
+from django.http import HttpResponse
 
 from urllib.parse import urlencode
+import csv
 
-from .models import Participant
+from .models import Participant, Attendance
 from .forms import CreateParticipantForm, NamiSearchForm
+from .templatetags.troop_extras import section_trans
 
 from nami import Nami, MemberNotFound
 
 
-class OnlyTroopManagerMixin(AccessMixin):
-    """Verify that the current user is allowed to manage the troop."""
+def user_managed_troop(user, troop_number):
```

Comment:
What do you think about implementing this on `User` or on `Troop`? Like `troop.is_managed_by(user)` or `user.manages(troop)`?

Reply:
It's a good idea, but I can't find the right implementation:
- adding it on the user model does not work for `AnonymousUser`
- adding it on `Troop` adds a useless db lookup.

I added it as a static method on the Troop class.

Reply:
Why does it not work for `AnonymousUser`? And one useless lookup wouldn't be too bad, right?

Reply:
> Why does it not work for AnonymousUser?

Because `AnonymousUser` does not inherit `models.User`, so I get a failure like `AttributeError: 'AnonymousUser' object has no attribute 'managed_troop'`

> And one useless lookup wouldn't be too bad, right?

Well, if it is useless, I would prefer not doing it...